spec/app/dispatcher_spec.rb

```diff
@@ -127,6 +127,21 @@
           it_behaves_like 'logs exception'
         end
 
+        context 'when app exits unexpectedly' do
+          let(:task) { :sync }
+
+          before(:each) do
+            allow(connector_settings).to receive(:service_type).and_return('')
+            allow(connector_settings).to receive(:index_name).and_return('')
+            allow(job_runner).to receive(:execute).and_raise(Interrupt)
+          end
+
+          it 'shuts down sync jobs' do
+            expect(described_class).to receive(:shutdown_sync_jobs_with_error)
+            expect { described_class.start! }.to_not raise_error
+          end
+        end
+
```

Comment:
Can you update other specs or add new ones to show that the pre/post hooks are operating correctly and adding/removing items from the cache as you expect?